authorizerd_test.go

```diff
@@ -1614,10 +1686,75 @@ func Test_authorizer_AuthorizeAccessToken(t *testing.T) {
 			}
 		}(),
 		func() test {
+			now := fastime.Now()
+			cert := &x509.Certificate{
+				Issuer: pkix.Name{
+					CommonName: "issuer cn",
+				},
+				Subject: pkix.Name{
+					CommonName: "subject cn",
+				},
+			}
 			c := gache.New()
-			c.Set("dummyTokdummyActdummyRes", &principal{})
+			at := &access.OAuth2AccessTokenClaim{}
+			p := &oAuthAccessToken{
+				principal: principal{
+					name:       at.BaseClaim.Subject,
+					roles:      at.Scope,
+					domain:     at.BaseClaim.Audience,
+					issueTime:  at.IssuedAt,
+					expiryTime: at.ExpiresAt,
+				},
+				clientID: at.ClientID,
+			}
+			c.SetWithExpire("dummyTok:"+cert.Issuer.CommonName+":"+cert.Subject.CommonName+":dummyAct:dummyRes", p, time.Minute)
 			apm := &AccessProcessorMock{
-				act:     &access.OAuth2AccessTokenClaim{},
+				atc:     at,
+				wantErr: nil,
+			}
+			pdm := &PolicydMock{
+				CheckPolicyFunc: func(ctx context.Context, domain string, roles []string, action, resource string) error {
+					if domain != "domain" || len(roles) != 1 || roles[0] != "role" {
+						return errors.New("Audience/Scope mismatch")
+					}
+					return nil
+				},
+			}
+			return test{
+				name: "test use cache success with cert",
+				args: args{
+					ctx:  context.Background(),
+					tok:  "dummyTok",
+					act:  "dummyAct",
+					res:  "dummyRes",
+					cert: cert,
+				},
+				fields: fields{
+					policyd:         pdm,
+					accessProcessor: apm,
+					cache:           c,
+					cacheExp:        time.Minute,
+				},
+				wantErr:    "",
+				wantResult: p,
+				checkFunc: func(prov *authority) error {
+					_, expiry, ok := prov.cache.GetWithExpire("dummyTok:issuer cn:subject cn:dummyAct:dummyRes")
+					if !ok {
```

Comment:
If you want to test that the cache is used, then how about checking the length of the cache?
```suggestion
					if !ok || prov.cache.Len() != 1 {
```
If you agree, please make the same change to line 1677.

Reply:
Fixed
https://github.com/yahoojapan/athenz-authorizer/pull/75/commits/d66340eeb7d1b116772ec9d30dd98d5a10fac43a